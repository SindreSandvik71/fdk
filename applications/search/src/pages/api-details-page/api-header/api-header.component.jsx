import React from 'react';
import PropTypes from 'prop-types';
import DocumentMeta from 'react-document-meta';
import Moment from 'react-moment';
import _ from 'lodash';

import localization from '../../../lib/localization';
import { DatasetLabelNational } from '../../../components/dataset-label-national/dataset-label-national.component';
import { getPublisherByOrgNr } from '../../../redux/modules/publishers';
import { getTranslateText } from '../../../lib/translateText';

export const ApiHeader = props => {
  const { title, publisher, harvest, publisherItems } = props;

  const renderPublisher = () => {
    if (!publisher) {
      return null;
    }
    const publisherItem = getPublisherByOrgNr(
      publisherItems,
      _.get(publisher, 'id')
    );

    const publisherPrefLabel =
      getTranslateText(_.get(publisherItem, ['prefLabel'])) ||
      _.capitalize(_.get(publisherItem, 'name', ''));

    if (publisherPrefLabel) {
      return (
        <div className="mb-4">
          <span>
            <span className="uu-invisible" aria-hidden="false">
              API
            </span>
            {localization.api.provider}&nbsp;
            <span className="fdk-strong-virksomhet">{publisherPrefLabel}</span>
          </span>
        </div>
      );
    }
  };

  const renderHarvested = () => {
    if (harvest && harvest.firstHarvested) {
      return (
        <span>
          {localization.dataset.firstHarvested}&nbsp;
          <Moment format="DD.MM.YYYY">{harvest.firstHarvested}</Moment>
        </span>
      );
    }
    return null;
  };

  const renderHarvestSeparator = () => {
    if (harvest && harvest.firstHarvested && harvest.lastChanged) {
      return <span>&nbsp;/&nbsp;</span>;
    }
    return null;
  };

  const renderLastChanged = () => {
    if (harvest && harvest.lastChanged) {
      return (
        <span>
          {localization.dataset.lastChanged}&nbsp;
          <Moment format="DD.MM.YYYY">{harvest.lastChanged}</Moment>
        </span>
      );
    }
    return null;
  };

  const meta = {
    title
  };
  return (
    <header>
      <DocumentMeta {...meta} />

      <div className="fdk-detail-date mb-4">
        {renderHarvested()}
        {renderHarvestSeparator()}
        {renderLastChanged()}
      </div>

      {title && (
        <div className="mb-2 d-flex flex-wrap align-items-baseline">
          <h1 className="mr-3">{title}</h1>
          <DatasetLabelNational />
        </div>
      )}

      <div className="mb-4">{renderPublisher()}</div>
    </header>
  );
};

ApiHeader.defaultProps = {
  title: null,
  publisher: null,
  harvest: null
};

ApiHeader.propTypes = {
  title: PropTypes.string,
  publisher: PropTypes.object,
  harvest: PropTypes.object
};
