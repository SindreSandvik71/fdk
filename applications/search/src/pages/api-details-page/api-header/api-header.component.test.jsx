import React from 'react';
import { shallow } from 'enzyme';
import { ApiHeader } from './api-header.component';
import header from './__fixtures/header';

let defaultProps;

test('should render ApiHeader with no props', () => {
  const wrapper = shallow(<ApiHeader />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ApiHeader correctly', () => {
  const wrapper = shallow(<ApiHeader {...header} />);
  expect(wrapper).toMatchSnapshot();
});
