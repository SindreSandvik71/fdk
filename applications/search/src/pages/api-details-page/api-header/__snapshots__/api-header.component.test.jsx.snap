// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ApiHeader correctly 1`] = `
<header>
  <SideEffect(DocumentMeta)
    title="Test"
  />
  <div
    className="fdk-detail-date mb-4"
  >
    <span>
      Published in The National Data Directory
       
      <t
        ago={false}
        calendar={false}
        decimal={false}
        element="time"
        filter={[Function]}
        format="DD.MM.YYYY"
        fromNow={false}
        interval={60000}
        onChange={[Function]}
        toNow={false}
        unit={null}
        unix={false}
        utc={false}
      >
        2018-06-25T19":51:05+0200
      </t>
    </span>
    <span>
       / 
    </span>
    <span>
      Last changed in The National Data Directory
       
      <t
        ago={false}
        calendar={false}
        decimal={false}
        element="time"
        filter={[Function]}
        format="DD.MM.YYYY"
        fromNow={false}
        interval={60000}
        onChange={[Function]}
        toNow={false}
        unit={null}
        unix={false}
        utc={false}
      >
        2018-07-09T14:40:37+0200
      </t>
    </span>
  </div>
  <div
    className="mb-2 d-flex flex-wrap align-items-baseline"
  >
    <h1
      className="mr-3"
    >
      Test
    </h1>
    <DatasetLabelNational />
  </div>
  <div
    className="mb-4"
  />
</header>
`;

exports[`should render ApiHeader with no props 1`] = `
<header>
  <SideEffect(DocumentMeta)
    title={null}
  />
  <div
    className="fdk-detail-date mb-4"
  />
  <div
    className="mb-4"
  />
</header>
`;
